gnd/src/main/java/com/google/android/gnd/ui/terms/TermsOfServiceViewModel.java

```diff
@@ -16,9 +16,31 @@
 
 package com.google.android.gnd.ui.terms;
 
+import androidx.lifecycle.MutableLiveData;
+import com.google.android.gnd.persistence.local.LocalValueStore;
+import com.google.android.gnd.rx.annotations.Hot;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import com.google.android.gnd.ui.common.Navigator;
+import javax.inject.Inject;
 
 // TODO: Needs to handle view state and behaviors of the Terms Fragment
 public class TermsOfServiceViewModel extends AbstractViewModel {
 
+  private final Navigator navigator;
+  private final LocalValueStore localValueStore;
+
+  @Hot(replays = true)
+  public final MutableLiveData<Boolean> termsCheckBox = new MutableLiveData<>();
+
+  @Inject
+  public TermsOfServiceViewModel(Navigator navigator, LocalValueStore localValueStore) {
+    this.navigator = navigator;
+    this.localValueStore = localValueStore;
+  }
+
+  public void onButtonClicked() {
+    localValueStore.setTermsAccepted(true);
```

Comment:
Rather than inject stores directly in the view models, please call through `TermsOfServiceManager`.

Reply:
already done in https://github.com/google/ground-android/pull/845